### PR TITLE
update orm with new async methods

### PR DESCRIPTION
**Pull request by tttmaximttt**

### CODE REVIEW

File: lib/ORM.js [hunk 1]
```diff
@@ -316,6 +319,28 @@ ORM.prototype.load = function () {
...
+  return Promise.each(filesArr, function (file) {
+    return new Promise(function (resolve, reject) {
+      var content = require(file);
+      content(self, function (err) {
+        if (err) reject(err);
+        resolve(file);
+      });
+    });
+  });
+};
```
Comment (dxg) [inline lib/ORM.js]: Why do we need to rewrite this code? Can't we `Promise.promisify` the original `load` function?

Comment (tttmaximttt) [reply lib/ORM.js]: no because this path resolve what written in original load not working with promising

Comment (dxg) [reply lib/ORM.js]: But why doesn't it work?

File: lib/ORM.js [hunk 2]
```diff
@@ -316,6 +319,28 @@ ORM.prototype.load = function () {
...
+  if (!Array.isArray(filesArr)) {
+    filesArr = [filesArr];
+  }
+
+  return Promise.each(filesArr, function (file) {
+    return new Promise(function (resolve, reject) {
+      var content = require(file);
+      content(self, function (err) {
+        if (err) reject(err);
+        resolve(file);
```
Comment (dxg) [inline lib/ORM.js]: We are calling both `reject` and `resolve`

File: lib/ORM.js (hunk 1 above)
Comment (tttmaximttt) [reply lib/ORM.js]: because the parsing file path is incorrect for me implement new method easy then read this code

Comment (dxg) [reply lib/ORM.js]: Sure but if we want to improve the code we should do it in a separate PR.
All we need here is to create a promise version. By rewriting code we create a high risk of new bugs, and different behaviour for callback & promise versions of the same function.

**Pull request merged**